### PR TITLE
firefox*: 107.0.1 -> 108.0; firefox-esr-unwrapped: 102.5.0esr -> 102.6.0esr

### PR DESCRIPTION
**Pull request by vcunat**
https://www.mozilla.org/en-US/firefox/108.0/releasenotes/
https://www.mozilla.org/en-US/firefox/102.6.0/releasenotes/

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vcunat** [comment]
I encountered an issue during build:
```
   Compiling proc-macro2 v1.0.43
     Running `CARGO=/nix/store/fkiykll6wmw34n5ynjwls6xpmjf1giv8-cargo-1.65.0/bin/.cargo-wrapped CARGO_CRATE_NAME=build_script_build CARGO_MANIFEST_DIR=/build/firefox-108.0/third_party/rust/proc-macro2 CARGO_PKG_AUTHORS='Davi
d Tolnay <dtolnay@gmail.com>:Alex Crichton <alex@alexcrichton.com>' CARGO_PKG_DESCRIPTION='A substitute implementation of the compiler'\''s `proc_macro` API to decouple token-based libraries from the procedural macro use cas
e.' CARGO_PKG_HOMEPAGE='' CARGO_PKG_LICENSE='MIT OR Apache-2.0' CARGO_PKG_LICENSE_FILE='' CARGO_PKG_NAME=proc-macro2 CARGO_PKG_REPOSITORY='https://github.com/dtolnay/proc-macro2' CARGO_PKG_RUST_VERSION=1.31 CARGO_PKG_VERSION
=1.0.43 CARGO_PKG_VERSION_MAJOR=1 CARGO_PKG_VERSION_MINOR=0 CARGO_PKG_VERSION_PATCH=43 CARGO_PKG_VERSION_PRE='' LD_LIBRARY_PATH='/build/firefox-108.0/mozobj/release/deps:/nix/store/f9b9jfn5qr9h2mg7jadl1dmm3b0ahc3w-rustc-1.65
.0/lib' /nix/store/f9b9jfn5qr9h2mg7jadl1dmm3b0ahc3w-rustc-1.65.0/bin/rustc --crate-name build_script_build --edition=2018 /build/firefox-108.0/third_party/rust/proc-macro2/build.rs --error-format=json --json=diagnostic-rende
red-ansi,artifacts,future-incompat --crate-type bin --emit=dep-info,link -C opt-level=1 -C embed-bitcode=no --cfg 'feature="default"' --cfg 'feature="proc-macro"' -C metadata=18500dc7a7ca1a9d -C extra-filename=-18500dc7a7ca1
a9d --out-dir /build/firefox-108.0/mozobj/release/build/proc-macro2-18500dc7a7ca1a9d -C linker=/build/firefox-108.0/build/cargo-linker -L dependency=/build/firefox-108.0/mozobj/release/deps --cap-lints warn`
make[3]: Leaving directory '/build/firefox-108.0/mozobj/memory/build'
make[3]: Entering directory '/build/firefox-108.0/mozobj/mozglue/baseprofiler'
mkdir -p '.deps/'
mozglue/baseprofiler/shared-libraries-linux.o
/nix/store/gf3v8vf6fk1ahc57fqfjv6rqgyiw4pxw-clang-wrapper-14.0.6/bin/clang++ -Qunused-arguments -std=gnu++17 -o shared-libraries-linux.o -c  -I/build/firefox-108.0/mozobj/dist/stl_wrappers -I/build/firefox-108.0/mozobj/dist/
system_wrappers -include /build/firefox-108.0/config/gcc_hidden.h -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -fstack-protector-strong -fstack-clash-protection -DNDEBUG=1 -DTRIMMED=1 -DIMPL_MFBT -DIMPL_MFBT -DMOZ_HAS_MOZGLUE -I/bu
ild/firefox-108.0/mozglue/baseprofiler -I/build/firefox-108.0/mozobj/mozglue/baseprofiler -I/build/firefox-108.0/mozglue/baseprofiler/core -I/build/firefox-108.0/mozglue/linker -I/build/firefox-108.0/mozobj/dist/include -I/n
ix/store/911xzw8fxivaj1cdcm53qvh8i1mpkff8-nspr-4.35-dev/include -I/nix/store/911xzw8fxivaj1cdcm53qvh8i1mpkff8-nspr-4.35-dev/include -I/nix/store/wkym4m118xqvsf6w0k2pn8v5wm3gcvl6-nss-3.85-dev/include/nss -I/build/firefox-108.
0/mozobj/dist/include/nss -I/nix/store/v4gn0w3rg3ci8mi5bzlsj3ls6hnm804i-zlib-1.2.13-dev/include -I/nix/store/6vbfxf54dj07yjffn4cz8sh7idbb7f33-libpng-apng-1.6.37-dev/include/libpng16 -I/nix/store/v4gn0w3rg3ci8mi5bzlsj3ls6hnm8
04i-zlib-1.2.13-dev/include -I/nix/store/nrzln1s5d9x2c3zis9aw1gfadq6g9f31-pixman-0.42.2/include/pixman-1 -I/nix/store/kycg24wmlwhlsdq01nwqmsvkmmq8kb0a-icu4c-72.1-dev/include -DMOZILLA_CLIENT -include /build/firefox-108.0/moz
obj/mozilla-config.h -fno-sized-deallocation -fno-aligned-new -fno-exceptions -fPIC -fno-rtti -ffunction-sections -fdata-sections -fno-exceptions -fno-math-errno -pthread -pipe -gdwarf-4 -O3 -fomit-frame-pointer -funwind-tab
les -Wall -Wbitfield-enum-conversion -Wdeprecated-this-capture -Wempty-body -Wformat-type-confusion -Wignored-qualifiers -Wpointer-arith -Wshadow-field-in-constructor-modified -Wsign-compare -Wtype-limits -Wno-error=tautolog
ical-type-limit-compare -Wunreachable-code -Wunreachable-code-return -Wunused-but-set-parameter -Wno-invalid-offsetof -Wclass-varargs -Wempty-init-stmt -Wfloat-overflow-conversion -Wfloat-zero-conversion -Wloop-analysis -Wno
-range-loop-analysis -Wc++2a-compat -Wenum-compare-conditional -Wenum-float-conversion -Wno-ambiguous-reversed-operator -Wno-error=deprecated -Wno-error=deprecated-anon-enum-enum-conversion -Wno-error=deprecated-enum-enum-co
nversion -Wno-error=deprecated-pragma -Wno-error=deprecated-this-capture -Wno-error=deprecated-volatile -Wcomma -Wimplicit-fallthrough -Wstring-conversion -Wno-inline-new-delete -Wno-error=deprecated-declarations -Wno-error=
array-bounds -Wno-error=free-nonheap-object -Wno-error=atomic-alignment -Wformat -Wformat-security -Wno-psabi -Wthread-safety -Wno-unknown-warning-option -Wno-ignored-qualifiers -fno-strict-aliasing -ffp-contract=off -DNS_FR
EE_PERMANENT_DATA=1 -fprofile-generate -MD -MP -MF .deps/shared-libraries-linux.o.pp   /build/firefox-108.0/mozglue/baseprofiler/core/shared-libraries-linux.cc
mozglue/baseprofiler/Unified_cpp_mozglue_baseprofiler0.o
error: linker `/build/firefox-108.0/build/cargo-linker` not found
  |
  = note: No such file or directory (os error 2)

error: could not compile `proc-macro2` due to previous error

Caused by:
  process didn't exit successfully: `CARGO=/nix/store/fkiykll6wmw34n5ynjwls6xpmjf1giv8-cargo-1.65.0/bin/.cargo-wrapped CARGO_CRATE_NAME=build_script_build CARGO_MANIFEST_DIR=/build/firefox-108.0/third_party/rust/proc-macro2 CARGO_PKG_AUTHORS='David Tolnay <dtolnay@gmail.com>:Alex Crichton <alex@alexcrichton.com>' CARGO_PKG_DESCRIPTION='A substitute implementation of the compiler'\''s `proc_macro` API to decouple token-based libraries from the procedural macro use case.' CARGO_PKG_HOMEPAGE='' CARGO_PKG_LICENSE='MIT OR Apache-2.0' CARGO_PKG_LICENSE_FILE='' CARGO_PKG_NAME=proc-macro2 CARGO_PKG_REPOSITORY='https://github.com/dtolnay/proc-macro2' CARGO_PKG_RUST_VERSION=1.31 CARGO_PKG_VERSION=1.0.43 CARGO_PKG_VERSION_MAJOR=1 CARGO_PKG_VERSION_MINOR=0 CARGO_PKG_VERSION_PATCH=43 CARGO_PKG_VERSION_PRE='' LD_LIBRARY_PATH='/build/firefox-108.0/mozobj/release/deps:/nix/store/f9b9jfn5qr9h2mg7jadl1dmm3b0ahc3w-rustc-1.65.0/lib' /nix/store/f9b9jfn5qr9h2mg7jadl1dmm3b0ahc3w-rustc-1.65.0/bin/rustc --crate-name build_script_build --edition=2018 /build/firefox-108.0/third_party/rust/proc-macro2/build.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type bin --emit=dep-info,link -C opt-level=1 -C embed-bitcode=no --cfg 'feature="default"' --cfg 'feature="proc-macro"' -C metadata=18500dc7a7ca1a9d -C extra-filename=-18500dc7a7ca1a9d --out-dir /build/firefox-108.0/mozobj/release/build/proc-macro2-18500dc7a7ca1a9d -C linker=/build/firefox-108.0/build/cargo-linker -L dependency=/build/firefox-108.0/mozobj/release/deps --cap-lints warn` (exit status: 1)
```

File: pkgs/applications/networking/browsers/firefox/common.nix [hunk 1]
```diff
@@ -235,6 +235,7 @@ buildStdenv.mkDerivation ({
   postPatch = ''
     rm -rf obj-x86_64-pc-linux-gnu
     patchShebangs mach
+    patchShebangs build
```
Comment (mweinelt) [inline pkgs/applications/networking/browsers/firefox/common.nix:238]: `patcheSheBangs` accepts multiple arguments.

**Comment by vcunat** [comment]
OK, it builds and starts up.

**Comment by mweinelt** [comment]
Tested the esr variant on x86_64-linux.

**Comment by vcunat** [comment]
I did build for me on `4ae626c8` (previous PR version).

File: pkgs/applications/networking/browsers/firefox/common.nix [hunk 2]
```diff
@@ -220,7 +220,7 @@ buildStdenv.mkDerivation ({
     "profilingPhase"
   ];
 
-  patches = lib.optionals (lib.versionOlder version "103") [
+  patches = lib.optionals (lib.versionOlder version "102.6.0") [
```
Comment (mweinelt) [inline pkgs/applications/networking/browsers/firefox/common.nix:223]: I removed this patch too early previously, that's when thunderbird failed to build.

**Comment by vcunat** [comment]
Release notes are up.

**Comment by vcunat** [comment]
So, I picked it atop 22.11 and have been using it without issues for most of the day.  Any more testers?  If this weren't a major update I would've merged it already.

**Comment by mweinelt** [comment]
Tested both on x86_64-linux on nixos unstable.

**Pull request merged**